fennel/testing/mock_client.py

```diff
@@ -134,14 +134,45 @@ def is_superset_featureset(featureset_new, featureset_old):
             )
             return features_new.issuperset(features_old)
 
+        def is_new_dataset_eligible(dataset_new, dataset_old):
+            """
+            This function check if dataset of same name comes in the increment mode,
+            is the new dataset eligible or not. Eligible if:
+            1. If version same and old == new then eligible.
+            2. new != old then version should be higher.
+            """
+            if dataset_new.version > dataset_old.version:
+                return True
+            elif dataset_new.version < dataset_old.version:
+                return False
+            else:
+                return dataset_old.signature() == dataset_new.signature()
```

Comment:
we were planning to deprecate signatures earlier, is there another check we can use or maybe define an equality instead of using signature ?

Reply:
would need equality at multiple places for the same, ie datasets, pipelines. Can take this task as another linear ticket.